superset-frontend/packages/superset-ui-chart-controls/src/utils/getBootstrapData.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { BootstrapData } from '../types';
+import { DEFAULT_BOOTSTRAP_DATA } from '../constants';
+
+export default function getBootstrapData(): BootstrapData {
```

Comment:
You don't need to replicate the bootstrap logic. You can define `row_limit` as:

``` typescript
const row_limit: SharedControlConfig<'SelectControl'> = {
   validators: [legacyValidateInteger, (v, state) => validateMaxValue(v, state?.maxValue)],
   mapStateToProps: state => ({ maxValue: state?.common.conf.SQL_MAX_ROW }),
   ...
}
```

You can remove all changes related to bootstrapping the configs.